### PR TITLE
migration exercise

### PR DESCRIPTION
**Pull request by Garvshah01**

### CODE REVIEW

File: db/migrate/20230405125701_change_default_enabled_on_products.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class ChangeDefaultEnabledOnProducts < ActiveRecord::Migration[7.0]
+  def change
+    change_column_default :products, :enabled , from:nil, to:false
```
Comment (shivamvinsol) [inline db/migrate/20230405125701_change_default_enabled_on_products.rb]: fix spacing

File: db/migrate/20230404045902_add_line_item_count_to_cart.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddLineItemCountToCart < ActiveRecord::Migration[7.0]
+  def change
+    add_column :carts, :line_item_count, :integer, null: false, default:0
```
Comment (shivamvinsol) [inline db/migrate/20230404045902_add_line_item_count_to_cart.rb]: spacing

File: db/migrate/20230404043931_add_column_to_products.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class AddColumnToProducts < ActiveRecord::Migration[7.0]
+  def change
+    add_column :products, :enabled, :boolean
+    add_column :products, :discount_price, :decimal, default:0
```
Comment (shivamvinsol) [inline db/migrate/20230404043931_add_column_to_products.rb]: spacing

File: db/migrate/20230404052334_irreversable_migration_example.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class IrreversableMigrationExample < ActiveRecord::Migration[7.0]
+  def change
+    create_table :articles do |t|
+      t.string :name
+      t.boolean :is_public?
```
Comment (shivamvinsol) [inline db/migrate/20230404052334_irreversable_migration_example.rb]: dont use `?` in db columns. Rename to `is_public`

File: db/migrate/20230404052334_irreversable_migration_example.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+    end
+    change_column_default :articles , :is_public? , true
```
Comment (shivamvinsol) [inline db/migrate/20230404052334_irreversable_migration_example.rb]: spacing issue
how is this example of irreversible migration?

File: app/views/products/_form.html.erb [hunk 6]
```diff
@@ -31,6 +31,21 @@
     <%= form.text_field :price %>
   </div>
 
+  <div>
+    <%= form.label :enabled, style: 'display: block'%>
+    <%= form.check_box :enabled%>
```
Comment (shivamvinsol) [inline app/views/products/_form.html.erb]: spacing

**Review by shivamvinsol — CHANGES_REQUESTED** [review]

File: db/migrate/20230404052334_irreversable_migration_example.rb (hunk 5 above)
Comment (Garvshah01) [reply db/migrate/20230404052334_irreversable_migration_example.rb]: because I don't provide from and to so :is_public never knows its previous state.

**Review by shivamvinsol — APPROVED** [review]

**Pull request merged**